nntrainer/tensor/tensor_pool.cpp

```diff
@@ -326,6 +329,26 @@ Tensor *TensorPool::extend(const std::string &name,
   return getTensor(name);
 }
 
+Tensor *TensorPool::createOrExtend(const std::string &name,
+                                   const TensorDim &dim,
+                                   const std::vector<unsigned int> &exec_order,
+                                   TensorLifespan lifespan,
+                                   const Tensor::Initializer &init) {
+  NNTR_THROW_IF(lifespan == TensorLifespan::UNMANAGED, std::invalid_argument)
+    << "unmanaged life span is not supported";
+
+  if (tensorExist(name)) {
+    Tensor *t = getTensor(name);
+    NNTR_THROW_IF(t->getDim() != dim, std::invalid_argument)
+      << "tensor dimension mismatch for createOrExtend name: " << name;
+    NNTR_THROW_IF(t->getInitializer() != init, std::invalid_argument)
+      << "tensor initializer mismatch for createOrExtend name: " << name;
+    return extend(name, exec_order, lifespan);
```

Comment:
use `requestPrerequestedTensor` in this case and not extend. Even all the checks being done here are done in `requestPrerequestedTensor`.

Reply:
This is createAndExtend(), `requestPrerequestedTensor` is `view`, could you explain more on this?

Reply:
As per offline discussion, extend() will have dimension check